crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -547,6 +560,108 @@ fn finalize_nominator_deposit<T: Config>(
+}
+
+fn slash_operator<T: Config>(
+    operator_id: OperatorId,
+    slash_info: PendingOperatorSlashInfo<NominatorId<T>, BalanceOf<T>>,
+) -> Result<BalanceOf<T>, TransitionError> {
+    Operators::<T>::try_mutate_exists(operator_id, |maybe_operator| {
+        // take the operator so this operator info is removed once we slash the operator.
+        let operator = maybe_operator
+            .take()
+            .ok_or(TransitionError::UnknownOperator)?;
+
+        let freeze_identifier = T::FreezeIdentifier::staking_freeze_id(operator_id);
+
+        let total_stake = operator
+            .current_total_stake
+            .checked_add(&operator.current_epoch_rewards)
+            .ok_or(TransitionError::BalanceOverflow)?;
+
+        let unlocking_withdrawals = slash_info
+            .unlocking_nominators
+            .into_iter()
+            .map(|pending_unlock| pending_unlock.balance)
+            .try_fold(BalanceOf::<T>::zero(), |acc, pending_withdrawal| {
+                acc.checked_add(&pending_withdrawal)
+                    .ok_or(TransitionError::BalanceOverflow)
+            })?;
+
+        let mut pending_deposits = BalanceOf::<T>::zero();
+
+        Nominators::<T>::drain_prefix(operator_id).try_for_each(|(nominator_id, _)| {
+            thaw_and_burn_frozen_balance::<T>(&freeze_identifier, &nominator_id)?;
+
+            // if there was a pending deposit from this nominator,
+            // then account that deposit in the slashed amount since this new deposit
+            // was burned above
+            if let Some(deposit) = PendingDeposits::<T>::take(operator_id, nominator_id) {
+                pending_deposits = pending_deposits
+                    .checked_add(&deposit)
+                    .ok_or(TransitionError::BalanceOverflow)?;
+            }
+
+            Ok(())
+        })?;
+
+        // remove all of the pending deposits since we burned all nominator stake.
+        let _ = PendingWithdrawals::<T>::clear_prefix(operator_id, u32::MAX, None);
+
+        // slash any new nominators as well
+        // all these are new nominators recorded after start
+        // of new epoch and before operator was slashed
+        for (nominator_id, deposit) in PendingDeposits::<T>::drain_prefix(operator_id) {
+            thaw_and_burn_frozen_balance::<T>(&freeze_identifier, &nominator_id)?;
+            pending_deposits = pending_deposits
+                .checked_add(&deposit)
+                .ok_or(TransitionError::BalanceOverflow)?;
+        }
+
+        // remove OperatorOwner Details
+        OperatorIdOwner::<T>::remove(operator_id);
+
+        // slashed_amount = current_total_stake + rewards + unlocking withdrawals + pending deposits
+        let slashed_amount = total_stake
+            .checked_add(&unlocking_withdrawals)
+            .and_then(|acc| acc.checked_add(&pending_deposits))
+            .ok_or(TransitionError::BalanceOverflow)?;
```

Comment:
I'm not sure if we should slash the total stake of the operator because it is expected (and we want it) to be large, if there are bugs (like taking invalid fraud proof as valid) or the operator is hacked then the operator will lose a huge amount of fortune, that may deter the operator from increasing its stake.

I remember at the offsite, we proposed to stop the operator from participating in staking upon detecting its first misbehavior to reduce its loss in case it is hacked.

Reply:
This was the proposed as part of the spec. Please start a discussion there